precompute-fileinfo/src/Main.hs

```diff
@@ -169,13 +169,17 @@ writeMap fp hashes = withFile fp WriteMode $ \h ->
 readMap :: FilePath -> IO (Map MD5 (SHA256, Length))
 readMap fp =
     withFile fp ReadMode $ \h -> do
-      hashes <- Map.fromList . map parseEntry . lines <$> hGetContents h
+      hashes <- mapFromParseEntry . lines <$> hGetContents h
       evaluate $ rnf hashes
       return hashes
   where
-    parseEntry :: String -> (MD5, (SHA256, Length))
-    parseEntry line = let [md5, sha256, len] = words line
-                      in (md5, (sha256, read len))
+    mapFromParseEntry :: [String] -> Map MD5 (SHA256, Length)
+    mapFromParseEntry mapLines = Map.fromList
+      [ case ws of
+          [md5, sha256, len] -> (md5, (sha256, read len))
+          _ -> (unwords ws, undefined)
+      | ws <- words <$> mapLines
+      ]
```

Comment:
If you just wanted to silence the `incomplete-uni-patterns` warning, the minimal change to the original code would be:
```haskell
parseEntry line =
  case words line of
     [md5, sha256, len] -> (md5, (sha256, read len))
     _ -> error $ "failed: parseEntry" ++ show line
```
It is rather obvious that this change does not otherwise the code.

Reply:
Yes minimal diff is better.